src/mono/System.Private.CoreLib/src/System/Runtime/DependentHandle.cs

```diff
@@ -0,0 +1,168 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.CompilerServices;
+
+namespace System.Runtime.CompilerServices
+{
+    internal struct Ephemeron
+    {
+        public object? key;
+        public object? value;
+    }
+}
+
+namespace System.Runtime
+{
+    //
+    // Instead of dependent handles, mono uses arrays of Ephemeron objects.
+    //
+    public struct DependentHandle : IDisposable
+    {
+        private Ephemeron[] data;
```

Comment:
All of the access to this appears to be guarded by null checks (unless I've missed a code path that doesn't).  Should this be `?`?  Then the `null!` in Dispose could lose the `!`.

Reply:
Done in f64d922a6000f6b9379048f8afaa8ec0b2a8d71c 👍